public/app/features/logs/components/panel/LogLineDetailsTrace.tsx

```diff
@@ -0,0 +1,111 @@
+import { css } from '@emotion/css';
+import { useEffect, useMemo, useState } from 'react';
+import { isObservable, lastValueFrom } from 'rxjs';
+
+import { DataFrame, DataSourceApi, GrafanaTheme2, TimeRange } from '@grafana/data';
+import { t } from '@grafana/i18n';
+import { getDataSourceSrv } from '@grafana/runtime';
+import { Icon, Spinner, Tooltip, useStyles2 } from '@grafana/ui';
+import { TraceView } from 'app/features/explore/TraceView/TraceView';
+import { transformDataFrames } from 'app/features/explore/TraceView/utils/transform';
+import { SearchTableType } from 'app/plugins/datasource/tempo/dataquery.gen';
+
+import { useLogListContext } from './LogListContext';
+import { EmbeddedInternalLink } from './links';
+
+interface Props {
+  traceRef: EmbeddedInternalLink;
+  timeRange: TimeRange;
+  timeZone: string;
+}
+
+export const LogLineDetailsTrace = ({ timeRange, timeZone, traceRef }: Props) => {
+  const [dataSource, setDataSource] = useState<DataSourceApi | null>(null);
+  const [dataFrames, setDataFrames] = useState<DataFrame[] | null | undefined>(undefined);
+  const { app } = useLogListContext();
+  const styles = useStyles2(getStyles);
+
+  useEffect(() => {
+    getDataSourceSrv()
+      .get(traceRef.dsUID)
+      .then((dataSource) => {
+        if (dataSource) {
+          setDataSource(dataSource);
+        } else {
+          setDataFrames(null);
+        }
+      });
+  }, [traceRef.dsUID]);
+
+  useEffect(() => {
+    if (!dataSource) {
+      return;
+    }
+    const query = dataSource.query({
+      app,
+      requestId: `log-details-trace-${traceRef.query}`,
+      targets: [
+        {
+          // @ts-expect-error
+          query: traceRef.query,
+          queryType: 'traceql',
+          refId: `log-details-trace-${traceRef.query}`,
+          tableType: SearchTableType.Traces,
+          filters: [],
+        },
+      ],
+      interval: '',
+      intervalMs: 0,
+      range: timeRange,
+      scopedVars: {},
+      timezone: timeZone,
+      startTime: Date.now(),
+    });
```

Comment:
If we type the query as a `DataQueryRequest<TempoQuery>` I don't think we need the ts ignore:

```suggestion
    const tempoQuery: DataQueryRequest<TempoQuery> = {
      app,
      requestId: 'test',
      targets: [
        {

          query: traceRef.query,
          queryType: 'traceql',
          refId: 'log-line-details-trace',
          tableType: SearchTableType.Traces,
          filters: [],
        },
      ],
      interval: '',
      intervalMs: 0,
      range: timeRange,
      scopedVars: {},
      timezone: timeZone,
      startTime: Date.now(),
    }
    const query = dataSource.query(tempoQuery);
```